builtin/providers/aws/resource_aws_s3_bucket.go

```diff
@@ -1310,17 +1315,30 @@ func removeNil(data map[string]interface{}) map[string]interface{} {
 	return withoutNil
 }
 
-func normalizeJson(jsonString interface{}) string {
+func normalizeJson(jsonString interface{}) (string, error) {
 	if jsonString == nil || jsonString == "" {
-		return ""
+		return "", nil
 	}
 	var j interface{}
 	err := json.Unmarshal([]byte(jsonString.(string)), &j)
 	if err != nil {
-		return fmt.Sprintf("Error parsing JSON: %s", err)
+		return "", err
```

Comment:
I think it's best if we return exactly what the user provided (`return jsonString.(string), err`) here so that `normalizeJsonStateFunc` will just return the value unchanged when it's invalid.

The reason for this is that `StateFunc` is called in some situations where `ValidateFunc` *isn't* called, such as normalizing a value set from the upstream API during a resource's `Read`, and so preserving the invalid value will allow it to show in a diff, rather than showing as the empty string.

This is definitely an edge case and mainly a cosmetic thing; I took a similar approach in #8350, which does something similar for normalizing/validating IAM policy JSON.